src/main/java/org/prebid/server/bidder/adview/AdviewBidder.java

```diff
@@ -0,0 +1,148 @@
+package org.prebid.server.bidder.adview;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.iab.openrtb.request.Banner;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.adview.ExtImpAdview;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class AdviewBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpAdview>> ADVIEW_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpAdview>>() {
+            };
+    private static final String ACCOUNT_ID_MACRO = "{{AccountId}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public AdviewBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final Imp firstImp = request.getImp().get(0);
+        final ExtImpAdview extImpAdview;
+
+        try {
+            extImpAdview = mapper.mapper().convertValue(firstImp.getExt(), ADVIEW_EXT_TYPE_REFERENCE).getBidder();
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
```

Comment:
Pls cover it with UT

Reply:
Ok, but should i cover it with UT? this should never happen since pbs core validates extension before it gets to this point.. maybe just get rid of this try catch?